shards/shards.go

```diff
@@ -702,6 +787,10 @@ func (s *shardedSearcher) getShards() []rankedShard {
 
 	var res []rankedShard
 	for _, sh := range s.shards {
+		// Hydrate indexData with visibility.
+		_, ok := s.public[sh.name]
+		sh.SetVisibility([]bool{ok})
```

Comment:
this is mutating data while searches can be running on loaded shards.

Reply:
I updated the PR to take the approach you proposed yesterday. The new apporach doesn't require mutating indexData.